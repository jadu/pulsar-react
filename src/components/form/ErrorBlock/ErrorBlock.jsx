import React from 'react';

export default class ErrorBlock extends React.Component {
  render() {
    if (!this.props.children) {
      return null;
    }

    let {
      children, 
      errorGuid,
      ...props} = this.props;

    return (
      <span id={errorGuid} className="help-block is-error">
        <i aria-hidden="true" className="icon-warning-sign"></i> {children}
      </span>
    );
  }
}
