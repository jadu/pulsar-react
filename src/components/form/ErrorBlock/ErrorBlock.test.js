'use strict';

import React from 'react';
import { shallow, render } from 'enzyme';

import ErrorBlock from './ErrorBlock';

it('Is null if no error is provided', () => {
  const tree = render(
    <ErrorBlock></ErrorBlock>
  );
  expect(tree).toMatchSnapshot();
});

it('Renders the help block when an error is provided', () => {
  const tree = render(
    <ErrorBlock>Foo</ErrorBlock>
  );
  expect(tree).toMatchSnapshot();
});

it('Renders the help block with errorGuid as the ID', () => {
  const tree = render(
    <ErrorBlock errorGuid="bar">Foo</ErrorBlock>
  );
  expect(tree).toMatchSnapshot();
});


