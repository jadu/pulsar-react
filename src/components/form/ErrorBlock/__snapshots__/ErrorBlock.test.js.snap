// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Is null if no error is provided 1`] = `null`;

exports[`Renders the help block when an error is provided 1`] = `
<span
  class="help-block is-error"
>
  <i
    aria-hidden="true"
    class="icon-warning-sign"
  />
   Foo
</span>
`;

exports[`Renders the help block with errorGuid as the ID 1`] = `
<span
  class="help-block is-error"
  id="bar"
>
  <i
    aria-hidden="true"
    class="icon-warning-sign"
  />
   Foo
</span>
`;
